### ISSUE TITLE
Updates on the rolling correlation

### ISSUE THREAD

**Issue by mindlessbrain** [man-group/dtale#50]
@mindlessbrain [v1.6.5](https://pypi.org/project/dtale/) is available which includes the rolling correlation functionality.  Will keep you posted on the other features I mentioned earlier.

_Originally posted by @aschonfeld in https://github.com/man-group/dtale/issues/45#issuecomment-569562470_

**Comment by mindlessbrain** [comment]
Hi, @aschonfeld.

As mentioned earlier, I am testing the rolling correlation functionality from the [release v1.6.5](https://pypi.org/project/dtale/), and I have some suggestions. This is all linked to previous issues #43 and #45. 

## Scatter Plot - improvements
1. The scatter is showing the full dataset. It should display only the data on the selected window, e.g.:

    - date: '2019-12-01'
    - window: 4
    - range: ['2019-11-28' : '2019-12-01']

    This is similar to pandas' index.iloc[:-window] but you'd also have to apply a logic for querying the specified date.

2. The tooltip on the scatter should show the date, not the index number 
![image](https://user-images.githubusercontent.com/17765738/71583213-38b11800-2aec-11ea-9e9e-70e459275bab.png)

3. The scatter plot title should also include the date range. This will be relevant once topic 1 is implemented.

## Line Plot  - bugs

1. I believe you are  not only expecting a datetime like column but also a column named as 'date'.
![image](https://user-images.githubusercontent.com/17765738/71583481-60ed4680-2aed-11ea-8f13-e7984b1ee94a.png)

2. It looks like it can't handle multi index
![image](https://user-images.githubusercontent.com/17765738/71583758-96defa80-2aee-11ea-97dc-1a4fe0714938.png)

The same errors happens for str multi indexes
![image](https://user-images.githubusercontent.com/17765738/71583842-e3c2d100-2aee-11ea-91ea-e6ef9b43644e.png)

![image](https://user-images.githubusercontent.com/17765738/71584341-efaf9280-2af0-11ea-858c-ec0d11c711d8.png)

## Code
```
import pandas as pd
import numpy as np
import dtale

ii = pd.date_range(start='2018-01-01', end='2019-12-01', freq='D')
ii = pd.Index(ii, name='date') 

n = ii.shape[0]
c = 5
data = np.random.random((n, c))

df = pd.DataFrame(data, index=ii)

d = dtale.show(df)

# change columns names
c = [(c, c) for c in df.columns]
df.columns = c
dtale.show(df)

# change columns with strings
c = [(str(c), str(c)) for c in range(5)]
df.columns = c
dtale.show(df)

# properly creating multi index
c = pd.MultiIndex.from_tuples(c)
df.columns = c
dtale.show(df)
```

**Comment by aschonfeld** [comment]
@mindlessbrain just finished work on a new version which includes stuff in the chart builder like:
- support for multi-column selection on the y-axis
- using rolling aggregations
- scatter charts

I will start working on these issues next, will keep you posted

**Comment by aschonfeld** [comment]
@mindlessbrain I have fixes coded for the line chart failures you were seeing.  Good catch on your part!  I'll work on re-working the scatter chart for the rolling correlations tomorrow morning.

**Comment by aschonfeld** [comment]
Fixed in [v1.6.7](https://github.com/man-group/dtale/releases/tag/v1.6.7)
